agb/src/display/tiled/vram_manager.rs

```diff
@@ -55,15 +60,50 @@ impl<'a> TileSet<'a> {
 }
 
 #[derive(Debug, Clone, Copy)]
-pub struct TileIndex(u16);
+pub enum TileIndex {
+    FourBpp(u16),
+    EightBpp(u8),
+}
 
 impl TileIndex {
-    pub(crate) const fn new(index: usize) -> Self {
-        Self(index as u16)
+    pub(crate) const fn new(index: usize, format: TileFormat) -> Self {
```

Comment:
maybe some `debug_assert!`s on the index to make sure we don't silently wrap? (probably more likely with the eightbpp version?)